kpi/management/commands/delete_assets_snapshots.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals, absolute_import
+
+from datetime import timedelta
+
+from django.db.models import Max
+from django.utils import timezone
+
+from .delete_base_command import BaseCommand
+from kpi.models import AssetSnapshot
+
+
+class Command(BaseCommand):
+
+    help = "Deletes assets snapshots"
+
+    def _prepare_delete_queryset(self, **options):
+        days = options["days"]
+        self._model = AssetSnapshot
+
+        # Retrieve Snapshots linked to assets' latest versions.
```

Comment:
(Sorry, I can't figure out how to highlight multiple lines for a comment, so I'm just highlighting the first one. This applies to lines 21 through 28)

Nice, but you can get the database to do all the work for you if you want:
```python
latest_version_ids = list(
    AssetSnapshot.objects.exclude(asset_version=None)
    .values('asset_id')
    .annotate(latest_version_id=Max('asset_version_id'))
    .values_list('latest_version_id', flat=True)
    .distinct()
)
```